presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateLongPercentileArrayAggregations.java

```diff
@@ -0,0 +1,120 @@
+
+    private ApproximateLongPercentileArrayAggregations() {}
+
+    @InputFunction
+    public static void arrayInput(DigestAndPercentileArrayState state, @SqlType(StandardTypes.BIGINT) long value, @SqlType("array<double>") Block quantilesArrayBlock)
+    {
+        QuantileDigest digest = state.getDigest();
+
+        if (digest == null) {
+            digest = new QuantileDigest(0.01);
+            state.setDigest(digest);
+            state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+        }
+
+        state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+        digest.add(value);
+        state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+
+        List<Double> percentiles = new ArrayList<>(quantilesArrayBlock.getPositionCount());
+
+        for (int i = 0; i < quantilesArrayBlock.getPositionCount(); i++) {
+            if (quantilesArrayBlock.isNull(i)) {
+                continue;
+            }
+            try {
+                percentiles.add(DOUBLE.getDouble(quantilesArrayBlock, i));
+            }
+            catch (Throwable t) {
+                Throwables.propagateIfInstanceOf(t, Error.class);
+                Throwables.propagateIfInstanceOf(t, PrestoException.class);
+
+                throw new PrestoException(INTERNAL_ERROR, t);
+            }
+        }
+
+        state.setPercentiles(percentiles);
+    }
+
+    @CombineFunction
+    public static void combine(DigestAndPercentileArrayState state, DigestAndPercentileArrayState otherState)
+    {
+        QuantileDigest input = otherState.getDigest();
+
+        QuantileDigest previous = state.getDigest();
+        if (previous == null) {
+            state.setDigest(input);
+            state.addMemoryUsage(input.estimatedInMemorySizeInBytes());
+        }
+        else {
+            state.addMemoryUsage(-previous.estimatedInMemorySizeInBytes());
+            previous.merge(input);
+            state.addMemoryUsage(previous.estimatedInMemorySizeInBytes());
+        }
+        state.setPercentiles(otherState.getPercentiles());
+    }
+
+    @OutputFunction("array<bigint>")
+    public static void output(DigestAndPercentileArrayState state, BlockBuilder out)
+    {
+        QuantileDigest digest = state.getDigest();
+        List<Double> percentiles = state.getPercentiles();
+
+        BlockBuilder blockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), percentiles.size());
+        for (int i = 0; i < percentiles.size(); i++) {
+            Double percentile = percentiles.get(i);
+            checkState(percentile != -1.0, "Percentile is missing");
+            checkCondition(0 <= percentile && percentile <= 1, INVALID_FUNCTION_ARGUMENT, "Percentile must be between 0 and 1");
+            try {
```

Comment:
Remove `try ... catch`. It's almost never a good idea to catch `NullPointerException` and swallow it.